### ISSUE TITLE
Add accessibility features to Drupal menus (primary, secondary, sidebar)

### ISSUE THREAD

**Issue by hyperboy** [SU-SWS/open_framework#22]
Feedback from Sean Keegan:

In general the OAA menu implementation is probably the best option (with a few caveats).  I like the Simply Accessible menu the best as it supports an intuitive interaction model (i.e., just using the arrow keys), but the problem I encountered is that it seemed to fail on the Mac platform with respect to screen-reader accessibility both with VoiceOver and Chrome+ChromeVox.  Like the author, though, I was only able to test in Snow Leopard...I have no test info for Lion at this time.

So, if I had to suggest one implementation, it would be the OAA Menubar model as these tended to work best on the Mac platform and do work for the Windows screen-readers as well.

The only big drawback on the OAA menubar model is that it tends to rely on some relatively newer coding, which could be an issue for individuals with older browsers.  It is identified as compliant with browsers as far back as Safari 5, FF3.6 and IE 8, so it's not that big an issue...but if someone is still rockin' IE 5.5, then there are no guarantees.

If you get a sample page implemented, drop me a line and I should have time to give it a test with a screen-reader.

**Comment by hyperboy** [comment]
- http://terrillthompson.com/blog/202
- http://staff.washington.edu/tft/tests/menus/html5/index.html

**Comment by meganem** [comment]
See accessibility review done by UComm recently.

**Comment by meganem** [comment]
1.  Need ARIA document landmarks on the page so screen reader users can jump easily to different sections of the page. Here is a description of the roles: http://juicystudio.com/article/examining-wai-aria-document-andmark-roles.php

The code to add a landmark is role=”main”, role=”contentinfo” and so on added to the div.

This should be marked with role=”banner”. The highlighted section should be marked with role=”search”.

This should be marked with role=”navigation”.

This should be marked with role=”contentinfo”.

This can be marked with role=”navigation”. It can be argued that this is also contentinfo. I prefer navigation in this case. There should be only one contentinfo role in a site.

This div should be marked up with role=”main”: div class="span9" id="content"

**Comment by hyperboy** [comment]
Added basic ARIA roles.  role="search" will need to be added to search block template.  Should be okay for now.
